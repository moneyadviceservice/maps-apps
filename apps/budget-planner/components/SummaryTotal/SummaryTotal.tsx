import { DetailedHTMLProps, HTMLAttributes, ReactNode } from 'react';

import { twMerge } from 'tailwind-merge';

import { Heading } from '@maps-react/common/components/Heading';
import {
  Props as SelectProps,
  Select,
} from '@maps-react/form/components/Select';
import { useTranslation } from '@maps-react/hooks/useTranslation';

export type Item = DetailedHTMLProps<
  HTMLAttributes<HTMLTableRowElement>,
  HTMLTableRowElement
> & {
  label: string;
  value: ReactNode;
  background?: string;
  isEstimate?: boolean;
};

const Item = ({
  label,
  value,
  className,
  background = 'bg-gray-100',
  isEstimate,
  ...rest
}: Item) => {
  return (
    <tr
      data-testid="summary-total-item-value"
      className={twMerge(
        `flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 ${background} ${
          isEstimate
            ? 'rounded-none border-8 border-y-0 rounded-bl-lg border-l-amber-500 text-black'
            : ''
        }`,
        className,
      )}
      {...rest}
    >
      <th className="text-lg font-normal">{label}</th>
      <td className="text-md" data-testid="item-value">
        {value}
      </td>
    </tr>
  );
};

export type Props = {
  children?: ReactNode;
  onSelectChange?: SelectProps['onChange'];
  selectOptions?: SelectProps['options'];
  items: Item[];
  summary: Item;
  defaultValue?: number;
  showHeading?: boolean;
  isProgress?: boolean;
};

export default function SummaryTotal({
  children,
  onSelectChange,
  defaultValue,
  selectOptions,
  items,
  summary,
  showHeading,
  isProgress,
}: Readonly<Props>) {
  const { z } = useTranslation();
  return (
    <div
      className={twMerge(
        `flex flex-col  print:border-none border border-slate-400 rounded-[4px] px-1 pb-4 ${
          !showHeading ? 'pt-3 pb-4' : ''
        }`,
      )}
    >
      {showHeading ? (
        <div className="flex items-center justify-center px-1 py-4 text-center md:flex-col lg:flex-row print:text-left print:p-2">
          <Heading level="h4" component="h3">
            {z({
              en: `${isProgress ? 'Your progress' : 'Summary total'}`,
              cy: `${isProgress ? 'Eich cynnydd' : 'Cyfanswm crynodeb'}`,
            })}
          </Heading>
          {isProgress && (
            <span className="ml-1 text-lg">
              {z({
                en: '(monthly)',
                cy: '(misol)',
              })}
            </span>
          )}
        </div>
      ) : null}
      {selectOptions && (
        <div className="px-6 pt-0 pb-4">
          <Select
            name="divisor"
            hideEmptyItem
            defaultValue={defaultValue ?? 1}
            onChange={onSelectChange}
            options={selectOptions}
          />
        </div>
      )}
      <table className="w-full border-collapse">
        <thead className="sr-only">
          <tr>
            <th className="hidden"></th>
            <th>Amount</th>
          </tr>
        </thead>
        <tbody className="flex flex-col mx-1">
          {items.map((item) => (
            <Item
              key={item.label}
              label={item.label}
              value={item.value}
              className={item.className}
              isEstimate={item.isEstimate}
            />
          ))}
          <Item
            label={summary.label}
            value={summary.value}
            background={summary.background}
            isEstimate={summary.isEstimate}
            className="text-white"
          />
        </tbody>
      </table>

      {children ? (
        <div className="flex justify-center px-1 mt-6 mb-4">{children}</div>
      ) : null}
    </div>
  );
}
