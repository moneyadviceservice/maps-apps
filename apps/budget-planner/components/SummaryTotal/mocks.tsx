import NumberFormat from '@maps-react/common/components/NumberFormat';

export const background = 'bg-green-700';
interface Item {
  label: string;
  value: JSX.Element;
  isEstimate: boolean;
}

const createItem = (
  label: string,
  value: number,
  direction?: 'increase' | 'decrease',
  isEstimate = false,
): Item => ({
  label,
  value: (
    <NumberFormat
      prefix="Â£"
      renderText={(value) => <b>{value}</b>}
      value={value}
      direction={direction}
    />
  ),
  isEstimate,
});

export const items: Item[] = [
  createItem('Income', 1000),
  createItem('Spending', 200),
];

export const itemsNeutral: Item[] = [
  createItem('Income', 1000),
  createItem('Spending', 1000),
];

export const itemsNegative: Item[] = [
  createItem('Income', 1000),
  createItem('Spending', Math.abs(-1200)),
];

export const itemsIncrease: Item[] = [
  createItem('Your income', 1400),
  createItem('Your budget now', 1100),
  createItem('Your new budget', 1100, 'increase'),
];

export const itemsDecrease: Item[] = [
  createItem('Your income', 1400),
  createItem('Your budget now', 1100),
  createItem('Your new budget', 980, 'decrease'),
];

export const itemsWithEstimate: Item[] = [
  createItem('Estimated Income', 1000, undefined, true),
  createItem('Estimated Spending', 200, undefined, true),
];
