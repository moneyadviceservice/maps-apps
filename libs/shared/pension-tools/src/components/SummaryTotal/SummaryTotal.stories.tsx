import { StoryFn } from '@storybook/react';

import SummaryTotal, { SUMMARY_TOTAL_STATUS_TYPES } from './SummaryTotal';

const StoryProps = {
  title: 'Components/PENSION-TOOLS/SummaryTotal',
  component: SummaryTotal,
  argTypes: {},
};

const Template: StoryFn<typeof SummaryTotal> = (args) => (
  <SummaryTotal {...args} />
);

export default StoryProps;
export const Default = Template.bind({});
const baseargs = {
  title: 'Summary total',
  income: 2000,
  spending: 1500,
  balance: 500,
  incomeLabel: 'Income',
  spendingLabel: 'Spending',
  balanceLabel: 'Balance',
};
Default.args = {
  ...baseargs,
  status: SUMMARY_TOTAL_STATUS_TYPES.BALANCED,
};

export const Positive = Template.bind({});
Positive.args = {
  ...baseargs,
  status: SUMMARY_TOTAL_STATUS_TYPES.POSITIVE,
};

export const Negative = Template.bind({});
Negative.args = {
  ...baseargs,
  status: SUMMARY_TOTAL_STATUS_TYPES.NEGATIVE,
};
