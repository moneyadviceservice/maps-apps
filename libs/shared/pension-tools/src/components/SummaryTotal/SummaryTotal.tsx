import { twMerge } from 'tailwind-merge';

import { H6 } from '@maps-react/common/components/Heading';
import NumberFormat from '@maps-react/common/components/NumberFormat';
import { Paragraph } from '@maps-react/common/components/Paragraph';

export enum SUMMARY_TOTAL_STATUS_TYPES {
  POSITIVE = 'positive',
  NEGATIVE = 'negative',
  BALANCED = 'balanced',
}

type Props = {
  title: string;
  income: number;
  spending: number;
  balance: number;
  incomeLabel: string;
  spendingLabel: string;
  balanceLabel: string;
  status?: SUMMARY_TOTAL_STATUS_TYPES;
  'data-testid'?: string;
};

export const SummaryTotal = ({
  'data-testid': testId,
  title,
  income,
  incomeLabel,
  spending,
  spendingLabel,
  balance,
  balanceLabel,
  status = SUMMARY_TOTAL_STATUS_TYPES.BALANCED,
}: Props) => {
  const mainClasses = [
    'py-[22px]',
    'pl-4',
    'pr-6',
    'rounded',
    'text-lg',
    'grid',
    'grid-cols-3',
  ];
  return (
    <div
      data-testid={testId}
      className="border border-slate-400 rounded px-2 pt-6 pb-4 h-fit space-y-2"
    >
      <H6 className="text-xl pl-2 pb-2">{title}</H6>
      {[
        { label: incomeLabel, value: income },
        { label: spendingLabel, value: spending },
      ].map((item, idx) => (
        <div key={idx} className={twMerge(mainClasses, 'bg-gray-100')}>
          <Paragraph className="col-span-2 m-0">{item.label}</Paragraph>
          <NumberFormat
            className="justify-self-end"
            prefix="£"
            value={item.value}
          />
        </div>
      ))}

      <div
        className={twMerge(
          mainClasses,
          'text-white',
          status === SUMMARY_TOTAL_STATUS_TYPES.BALANCED && 'bg-gray-650',
          status === SUMMARY_TOTAL_STATUS_TYPES.POSITIVE && 'bg-green-700',
          status === SUMMARY_TOTAL_STATUS_TYPES.NEGATIVE && 'bg-red-700',
        )}
      >
        <Paragraph className="col-span-2 m-0">{balanceLabel}</Paragraph>
        <NumberFormat className="justify-self-end" prefix="£" value={balance} />
      </div>
    </div>
  );
};
