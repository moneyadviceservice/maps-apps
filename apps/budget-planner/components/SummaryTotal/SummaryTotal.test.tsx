import NumberFormat from '@maps-react/common/components/NumberFormat';
import { render } from '@testing-library/react';
import {
  background,
  items,
  itemsDecrease,
  itemsIncrease,
  itemsNegative,
  itemsNeutral,
  itemsWithEstimate,
} from './mocks';
import SummaryTotal from './SummaryTotal';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('SummaryTotal component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SummaryTotal
        items={items}
        summary={{
          label: 'Spare cash',
          value: (
            <NumberFormat
              prefix="£"
              renderText={(value) => <b>{value}</b>}
              value={800}
            />
          ),
          background,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with children', () => {
    const { container } = render(
      <SummaryTotal
        items={items}
        summary={{
          label: 'Spare cash',
          value: (
            <NumberFormat
              prefix="£"
              renderText={(value) => <b>{value}</b>}
              value={800}
            />
          ),
          background,
        }}
      >
        <button>Reset the calculator</button>
      </SummaryTotal>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with dropdown', () => {
    const { container } = render(
      <SummaryTotal
        items={items}
        summary={{
          label: 'Spare cash',
          value: (
            <NumberFormat
              prefix="£"
              renderText={(value) => <b>{value}</b>}
              value={800}
            />
          ),
          background,
        }}
        selectOptions={[
          { value: '1', text: 'Per month' },
          { value: '12', text: 'Per year' },
        ]}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders neutral', () => {
    const { container } = render(
      <SummaryTotal
        items={itemsNeutral}
        summary={{
          label: 'Zero balance',
          value: (
            <NumberFormat
              prefix="£"
              renderText={(value) => <b>{value}</b>}
              value={0}
            />
          ),
          background: 'bg-gray-400',
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders negative', () => {
    const { container } = render(
      <SummaryTotal
        items={itemsNegative}
        summary={{
          label: 'Overspend',
          value: (
            <NumberFormat
              prefix="£"
              renderText={(value) => <b>{value}</b>}
              value={200}
            />
          ),
          background: 'bg-red-600',
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with increase', () => {
    const { container } = render(
      <SummaryTotal
        items={itemsIncrease}
        summary={{
          label: 'Spare cash',
          value: (
            <NumberFormat
              prefix="£"
              renderText={(value) => <b>{value}</b>}
              value={800}
            />
          ),
          background,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with decrease', () => {
    const { container } = render(
      <SummaryTotal
        items={itemsDecrease}
        summary={{
          label: 'Spare cash',
          value: (
            <NumberFormat
              prefix="£"
              renderText={(value) => <b>{value}</b>}
              value={800}
            />
          ),
          background,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with Estimate', () => {
    const { container } = render(
      <SummaryTotal
        items={itemsWithEstimate}
        summary={{
          isEstimate: true,
          label: 'Estimated Spare Cash',
          value: (
            <NumberFormat
              prefix="£"
              renderText={(value) => <b>{value}</b>}
              value={800}
            />
          ),
          background,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
