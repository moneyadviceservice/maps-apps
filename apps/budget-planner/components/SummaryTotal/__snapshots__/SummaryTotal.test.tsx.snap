// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryTotal component renders correctly 1`] = `
<div
  class="flex flex-col print:border-none border border-slate-400 rounded-[4px] px-1 pt-3 pb-4"
>
  <table
    class="w-full border-collapse"
  >
    <thead
      class="sr-only"
    >
      <tr>
        <th
          class="hidden"
        />
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="flex flex-col mx-1"
    >
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Income
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,000
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spending
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £200
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-green-700 text-white"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spare cash
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £800
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`SummaryTotal component renders negative 1`] = `
<div
  class="flex flex-col print:border-none border border-slate-400 rounded-[4px] px-1 pt-3 pb-4"
>
  <table
    class="w-full border-collapse"
  >
    <thead
      class="sr-only"
    >
      <tr>
        <th
          class="hidden"
        />
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="flex flex-col mx-1"
    >
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Income
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,000
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spending
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,200
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-red-600 text-white"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Overspend
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £200
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`SummaryTotal component renders neutral 1`] = `
<div
  class="flex flex-col print:border-none border border-slate-400 rounded-[4px] px-1 pt-3 pb-4"
>
  <table
    class="w-full border-collapse"
  >
    <thead
      class="sr-only"
    >
      <tr>
        <th
          class="hidden"
        />
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="flex flex-col mx-1"
    >
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Income
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,000
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spending
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,000
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-400 text-white"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Zero balance
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £0
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`SummaryTotal component renders with Estimate 1`] = `
<div
  class="flex flex-col print:border-none border border-slate-400 rounded-[4px] px-1 pt-3 pb-4"
>
  <table
    class="w-full border-collapse"
  >
    <thead
      class="sr-only"
    >
      <tr>
        <th
          class="hidden"
        />
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="flex flex-col mx-1"
    >
      <tr
        class="flex justify-between print:border-none w-full h-full mb-2 py-5 px-4 bg-gray-100 rounded-none border-8 border-y-0 rounded-bl-lg border-l-yellow-700 text-black"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Estimated Income
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,000
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none w-full h-full mb-2 py-5 px-4 bg-gray-100 rounded-none border-8 border-y-0 rounded-bl-lg border-l-yellow-700 text-black"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Estimated Spending
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £200
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none w-full h-full mb-2 py-5 px-4 bg-green-700 rounded-none border-8 border-y-0 rounded-bl-lg border-l-yellow-700 text-white"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Estimated Spare Cash
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £800
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`SummaryTotal component renders with children 1`] = `
<div
  class="flex flex-col print:border-none border border-slate-400 rounded-[4px] px-1 pt-3 pb-4"
>
  <table
    class="w-full border-collapse"
  >
    <thead
      class="sr-only"
    >
      <tr>
        <th
          class="hidden"
        />
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="flex flex-col mx-1"
    >
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Income
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,000
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spending
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £200
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-green-700 text-white"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spare cash
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £800
          </b>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="flex justify-center px-1 mt-6 mb-4"
  >
    <button>
      Reset the calculator
    </button>
  </div>
</div>
`;

exports[`SummaryTotal component renders with decrease 1`] = `
<div
  class="flex flex-col print:border-none border border-slate-400 rounded-[4px] px-1 pt-3 pb-4"
>
  <table
    class="w-full border-collapse"
  >
    <thead
      class="sr-only"
    >
      <tr>
        <th
          class="hidden"
        />
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="flex flex-col mx-1"
    >
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Your income
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,400
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Your budget now
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,100
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Your new budget
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <span
            class="inline-flex"
          >
            <span
              class="flex items-center gap-2 flex-row-reverse"
            >
              <b>
                £980.00
              </b>
              <span
                class="inline-block border-solid border-x-transparent border-t-[13px] border-x-[8px] border-b-0 border-t-green-700"
              />
            </span>
          </span>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-green-700 text-white"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spare cash
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £800
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`SummaryTotal component renders with dropdown 1`] = `
<div
  class="flex flex-col print:border-none border border-slate-400 rounded-[4px] px-1 pt-3 pb-4"
>
  <div
    class="px-6 pt-0 pb-4"
  >
    <div
      class="relative block focus-within:border-purple-600 border-transparent border focus-within:rounded focus-within:shadow-select-focus"
    >
      <span
        class="absolute inset-y-0 right-0 flex items-center w-10 pl-1 pr-1 text-white bg-pink-600 rounded-r pointer-events-none print:hidden shadow-bottom-gray focus:shadow-none"
      >
        <svgr-mock
          classname="h-full"
        />
      </span>
      <select
        class="print:p-1 text-gray-500 border-gray-400 border-y border-l w-full outline-none block text-md h-10 pl-3 pr-12 bg-white rounded focus:border-white tool-dd"
        name="divisor"
      >
        <option
          selected=""
          value="1"
        >
          Per month
        </option>
        <option
          value="12"
        >
          Per year
        </option>
      </select>
    </div>
  </div>
  <table
    class="w-full border-collapse"
  >
    <thead
      class="sr-only"
    >
      <tr>
        <th
          class="hidden"
        />
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="flex flex-col mx-1"
    >
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Income
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,000
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spending
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £200
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-green-700 text-white"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spare cash
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £800
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`SummaryTotal component renders with increase 1`] = `
<div
  class="flex flex-col print:border-none border border-slate-400 rounded-[4px] px-1 pt-3 pb-4"
>
  <table
    class="w-full border-collapse"
  >
    <thead
      class="sr-only"
    >
      <tr>
        <th
          class="hidden"
        />
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="flex flex-col mx-1"
    >
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Your income
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,400
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Your budget now
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £1,100
          </b>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-gray-100"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Your new budget
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <span
            class="inline-flex"
          >
            <span
              class="flex items-center gap-2 flex-row-reverse"
            >
              <b>
                £1,100.00
              </b>
              <span
                class="inline-block border-solid border-x-transparent border-b-[13px] border-x-[8px] border-t-0 border-b-red-700"
              />
            </span>
          </span>
        </td>
      </tr>
      <tr
        class="flex justify-between print:border-none border w-full h-full rounded-[4px] mb-2 py-5 px-4 bg-green-700 text-white"
        data-testid="summary-total-item-value"
      >
        <th
          class="text-lg font-normal"
        >
          Spare cash
        </th>
        <td
          class="text-md"
          data-testid="item-value"
        >
          <b>
            £800
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
